modin/numpy/arr.py

```diff
@@ -190,6 +205,38 @@ def __init__(
             {col_name: new_dtype for col_name in self._query_compiler.columns}
         )
 
+    def _add_sibling(self, sibling):
```

Comment:
can we re-use that function from `BasePandasDataset` instead of copy-pasting it here somehow?

Reply:
Spoke to @devin-petersohn and he said he didn't like mixins + it would be more code to add it to a mixin or inheritance class, so he recommends just leaving it!

Reply:
I can see a point of "don't like", but it would actually be less code - you'd cut it from `BasePandasDataset` and paste somewhere else, and inherit in two places.

Technically it can even be not a mixin, but a common ancestor class for `BasePandasDataset` and this `array`, would it be better?

Reply:
That makes sense in the long run, but I think for now, we want to avoid that architectural complexity.